From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SoSeDiK <mrsosedik@gmail.com>
Date: Sun, 23 Jun 2024 15:43:34 +0300
Subject: [PATCH] Allow riders to control horses without a saddle

Requires a saddle on the client's side to actually control the entity.
This is handled by a plugin on SoSeDiK's Universe.

diff --git a/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java b/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java
index f58a0f50d04004587d342c1bb5f681cd485cf302..343ad5bb780b15d7d90bb4c38a4ea114761b2fcb 100644
--- a/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java
+++ b/src/main/java/net/minecraft/world/entity/animal/horse/AbstractHorse.java
@@ -228,8 +228,8 @@ public abstract class AbstractHorse extends Animal implements ContainerListener,
 
     protected AbstractHorse(EntityType<? extends AbstractHorse> type, Level world) {
         super(type, world);
-        this.moveControl = new net.minecraft.world.entity.ai.control.MoveControl(this); // Purpur - use vanilla controller
-        this.lookControl = new net.minecraft.world.entity.ai.control.LookControl(this); // Purpur - use vanilla controller
+        // this.moveControl = new net.minecraft.world.entity.ai.control.MoveControl(this); // Purpur - use vanilla controller // Kiterino - Allow riders to control horses without a saddle
+        // this.lookControl = new net.minecraft.world.entity.ai.control.LookControl(this); // Purpur - use vanilla controller // Kiterino - Allow riders to control horses without a saddle
         this.createInventory();
     }
 
@@ -400,7 +400,7 @@ public abstract class AbstractHorse extends Animal implements ContainerListener,
 
     @Override
     public boolean isSaddled() {
-        return this.getFlag(4);
+        return this.getFlag(4) || hasRider(); // Kiterino - Allow riders to control horses without a saddle
     }
 
     public int getTemper() {
@@ -592,7 +592,7 @@ public abstract class AbstractHorse extends Animal implements ContainerListener,
 
     @Override
     public void openCustomInventoryScreen(Player player) {
-        if (!this.level().isClientSide && (!this.isVehicle() || this.hasPassenger((Entity) player)) && this.isTamed()) {
+        if (!this.level().isClientSide && (!this.isVehicle() || this.hasPassenger((Entity) player)) && (this.isTamed() || getRider() == player)) { // Kiterino - Allow riders to control horses without a saddle
             player.openHorseInventory(this, this.inventory);
         }
 
